src/frontend/src/optimizer/plan_node/logical_project_set.rs

```diff
@@ -400,6 +400,17 @@ impl ToStream for LogicalProjectSet {
     // TODO: implement to_stream_with_dist_required like LogicalProject
 
     fn to_stream(&self, ctx: &mut ToStreamContext) -> Result<PlanRef> {
+        if self.select_list().iter().any(|item| item.has_now()) {
+            // User may use `now()` in table function in a wrong way, because we allow `now()` in `FROM` clause.
+            return Err(ErrorCode::NotSupported(
+                "General `now()` function".to_string(),
+                "If you are trying to use `generate_series` with `now()`, please kindly check \
+                whether you are using it in a correct way."
```

Comment:
Do we still need this hint if we have noticed (warned) user during executing the optimization rule?

Reply:
But we have to provide a hint for `NotSupported` error😄